airflow/serialization/pydantic/dag_run.py

```diff
@@ -30,7 +29,7 @@ class DagRunPydantic(BaseModelPydantic):
     id: int
     dag_id: str
     queued_at: Optional[datetime]
-    execution_date: DateTime
+    execution_date: datetime
```

Comment:
Ah... So Pydantic just got more "picky" about those. Cool.

Reply:
Yeah I think they basically switch from `isinstance` to type identity check, probably for performance concerns (we have similar designs in our own serialisation code). The old behaviour can be brought back with a flag, but this doesn’t need that.